jerry-core/mem/mem-poolman.cpp

```diff
@@ -86,61 +165,325 @@ mem_pools_init (void)
 void
 mem_pools_finalize (void)
 {
-  JERRY_ASSERT (mem_pools == NULL);
+  mem_pools_collect_empty ();
+
+#ifndef JERRY_NDEBUG
   JERRY_ASSERT (mem_free_chunks_number == 0);
+#endif /* !JERRY_NDEBUG */
 } /* mem_pools_finalize */
 
+void
+mem_pools_collect_empty (void)
+{
+  /*
+   * Hint magic number in header of pools with free first chunks
+   */
+  const uint16_t hint_magic_num_value = 0x7e89;
```

Comment:
@zherczeg, I've just used `cat /dev/random | hexdump -C` and took first number from the output.
`0` can appear more frequently as it is generally used for initialization of memory locations, so just random number would be more suitable in the case.